SimpleChurchAPI.php

```diff
@@ -10,15 +10,15 @@ class SimpleChurchAPI
 
     public function __construct($opts = array())
     {
-        if ($opts['sessionId']) {
-            $this->setSessionId($opts['sessionId']);
-        }
-
-        if (!$opts['subDomain']) {
+        if (!empty($opts['subDomain'])) {
```

Comment:
We can simplify this without the else via:

```
if (empty($opts['subDomain'])) {
    throw new Exception('subDomain is required.');
}

$this->setSubDomain($opts['subDomain']);
```

Also I'm not a fan of a required field being in a variable named `$opts` which implies optional (not related to your changes, just made me chuckle).

Reply:
I actually didn't see it necessary to accept an array as an argument. In the next two commits, they are separate arguments.

Reply:
that will break existing clients and is less future proof. lets keep it an array. feel free to rename as "args" or something.

Reply:
Sounds good, in that case I will keep things as-is but tidy up the constructor.